src/pre_process/m_initial_condition.fpp

```diff
@@ -283,6 +285,1204 @@ contains
         end do
 
     end subroutine s_perturb_surrounding_flow ! ----------------------------
+	
+	subroutine s_superposition_instability_wave() ! ------------------------------
```

Comment:
does this only work in 2D or 3D? if so, this should be added to the startup checks in `m_checker.f90` (or whatever the file is called) so that you cannot run `instability_wave` on a 1D problem.

Reply:
Yes, this is only work in 2D and 3D. I added checks in m_checker.f90 to prevent running on a 1D case.